cmake/CheckGLM.cmake

```diff
@@ -1,14 +1,23 @@
+set(GLM_WORKS 0)
 include(CheckIncludeFileCXX)
 check_include_file_cxx("glm/glm.hpp" HAVE_GLM)
-if(NOT HAVE_GLM)
-    unset(HAVE_GLM CACHE)
-    message(FATAL_ERROR "GLM was not found. Please install it from your repositories (usually named libglm-dev or glm-devel), or from https://glm.g-truc.net .")
+if(HAVE_GLM)
+	message(STATUS "Checking that GLM has the required features")
+	# constexpr works since GLM 0.9.9.5.
+	try_compile(HAVE_GLM_CONSTEXPR "${CMAKE_BINARY_DIR}/detect/"
+				"${CMAKE_CURRENT_LIST_DIR}/check-glm-constexpr-ctor.cpp")
+	if(HAVE_GLM_CONSTEXPR)
+		set(GLM_WORKS 1)
+	endif()
 endif()
-message(STATUS "Checking that GLM has the required features")
-try_compile(HAVE_GLM_CONSTEXPR "${CMAKE_BINARY_DIR}/detect/"
-            "${CMAKE_CURRENT_LIST_DIR}/check-glm-constexpr-ctor.cpp")
-if(NOT HAVE_GLM_CONSTEXPR)
-    unset(HAVE_GLM_CONSTEXPR CACHE)
-    message(FATAL_ERROR "GLM doesn't appear to support constexpr constructors. Please check that your GLM is at least 0.9.9.5.")
+unset(HAVE_GLM CACHE)
+unset(HAVE_GLM_CONSTEXPR CACHE)
+
+if(GLM_WORKS)
+	message(STATUS "Checking that GLM has the required features - done")
+	add_library(glm::glm IMPORTED INTERFACE)
+else()
+	CPMAddPackage(NAME glm
```

Comment:
Silent downloading even if _some_ GLM is present is not good. If the user has some version of GLM installed, they may be in power to upgrade it. Failure of `check-glm-constexpr-ctor` should result in a message as it used to.
Also, can we ask CPM not to download and just fail by default (so that the download only happened when corresponding CMake option was passed)? I really dislike it like when a build system downloads something without my approval.

Reply:
> should result in a message as it used to.

Done

> Also, can we ask CPM not to download and just fail by default (so that the download only happened when corresponding CMake option was passed)?

That kinda defeats the purpose. But `CPM_USE_LOCAL_PACKAGES=true` or `CPM_LOCAL_PACKAGES_ONLY=true` would do it. They both default to false. They should be set while packaging for a distro, where downloading something without approval won't even work. I wouldn't set them to true here, I think it should be user choice.

Note that I'm trying to sit on 2 chairs at once: to make this packageable as Gentoo maintainer (it's packageable already), and to make its usage in stellarium outside of any distro easy for Stellarium devs. These goals are conflicting, and CPM just allows achieving both.

PS. the gentoo package doesn't require this PR, it only waits for some other blocker of Stellarium (QtCharts from Qt 6 is not available in gentoo yet)

Reply:
I've added CPM_LOCAL_PACKAGES_ONLY to appveyor config for linux build